### PR TITLE
fix: Add spark to lambda dockerfile

### PR DESCRIPTION
**Pull request by achals**
Signed-off-by: Achal Shah <achals@gmail.com>

**What this PR does / why we need it**:

There were two issues with the AWS Lambda feature server tests:
- The image would fail to start because of missing pyspark dependencies:
```
[ERROR] Runtime.ImportModuleError: Unable to import module 'app': No module named 'pyspark' Traceback (most recent call last):
```

This was temporarily fixed by adding pyspark to the image used for Lambda. Note that this should be removed as soon as possible since the image size doubled as a result of adding this dependency.

- After fixing the previous issue, we'd see this kind of error:
```
Exception: Failed to get online features from remote feature server ***'detail': 'Failed to parse entities field: from_json_object() takes 3 positional arguments but 4 were given.'
```
This was traced to an unpinned version of protobuf - `3.20.0` changed the API in the proto_json module, which we were patching in the feature server. We've temporarily constrained the upper bound of protobuf used but should follow up with an upgrade to the version of protobuf along with a fix to the patched version.

**Which issue(s) this PR fixes**:

Fixes #

### CODE REVIEW

File: sdk/python/feast/proto_json.py [hunk 1]
```diff
@@ -140,10 +140,12 @@ def to_json_object(printer: _Printer, message: ProtoMessage) -> JsonObject:
         return [printer._MessageToJsonObject(item) for item in message.val]
 
     def from_json_object(
-        parser: _Parser, value: JsonObject, message: ProtoMessage
+        parser: _Parser, value: JsonObject, message: ProtoMessage,
     ) -> None:
+        print("going to convert message")
```
Comment (felixwang9817) [inline sdk/python/feast/proto_json.py:145]: nit: remove `print`

File: sdk/python/setup.py [hunk 2]
```diff
@@ -52,7 +52,7 @@
     "mmh3",
     "pandas>=1.0.0",
     "pandavro==1.5.*",
-    "protobuf>=3.10",
+    "protobuf>=3.10,<3.20",
```
Comment (felixwang9817) [inline sdk/python/setup.py:55]: nit: do we want to also update `py3.7-requirements.txt` and other files?

**Review by felixwang9817 — CHANGES_REQUESTED** [review]

File: sdk/python/feast/proto_json.py (hunk 1 above)
Comment (achals) [reply sdk/python/feast/proto_json.py:145]: good catch

File: sdk/python/setup.py (hunk 2 above)
Comment (achals) [reply sdk/python/setup.py:55]: oh yeah good point

Comment (achals) [reply sdk/python/setup.py:55]: just read through what the `make lock-python-*dependencies` command does and existing requirements file - I don't think any changes are needed here.

**Review by felixwang9817 — APPROVED** [review]
/lgtm

**Pull request merged**